DM-21920 Add notebook to demonstrate DFT properties

kgabor opened the pull request.

Commit:
Added notebook to demonstrate points in the ticket.

isullivan reviewed (changes requested):
General comments:
Please do another round of proofreading for typos in the comments. Also, there are many incomplete sentences, which often leaves the meaning ambiguous.

Please add sections to the code examples, and add numbers to keep track of them. Then you can add references to those sections in the commentary and conclusions at the start. It probably wouldn't hurt to repeat the text of the relevant "Conclusions" bullet point at the end of the code section where you demonstrate it. 

Some specific comments on different sections:

"Some Properties of DFT"
- Item 2 is full of incomplete sentences, that make the reader guess at your meaning
- scr_2d_dft_freqs.png image does not display on Github

"Convolution theorem in discrete case"
- What do you mean by "circular convolution"? Are you referring to it being periodic?

"Conclusions"
- This sentence is unclear: "Also, after some Fourier space operations on FT(padded kernel), the result usually cannot be transformed back to real space as a small image but will remain nonzero accross the whole image."
What do you mean by "...as a small image but..."

Commit:
Review response: clarification of notebook text.

kgabor commented:
- I've updated the text to be more clear.
- I don't think the absence of the figure in github can be directly rectified. I've added a link that should point to the raw image file once it is merged into master.

isullivan commented:
This is much easier to follow, thank you. A few minor comments on the updated version:
In section 5 you refer to "Therefore we need to apply the zero padding of 4.3", but I think you mean 4.2?
You also refer to 4.3 in 6.3.
Please add some additional commentary to section 11, the decorrelation afterburner correction. I think a single sentence pointing out that it is much smoother with padding would be sufficient, unless you think more should be said about the pattern visible in pkdecorr2 of the final cell.
